src/FileServiceManager/FileService/DAL/Storages/FileSystemStorage/FileSystemStorage.cs

```diff
@@ -0,0 +1,132 @@
+﻿using FileService.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using FileService.Helpers;
+
+namespace FileService.DAL
+{
+    public class FileSystemStorage : IStorage
+    {
+        private readonly string _basePath;
+        private readonly int _fileSizeLimit;
+        private readonly IEnumerable<string> _allowedExtensions;
+        private readonly IFileSystemConfigurator _fileSystemConfigurator;
+
+        public FileSystemStorage(IConfiguration configuration, IFileSystemConfigurator fileSystemConfigurator)
+        {
+            _basePath = Path.Combine((configuration["storage:path"] ?? Directory.GetCurrentDirectory()) + "\\Files\\");
+
+            if (!int.TryParse(configuration["file:sizeLimit"], out int fileSizeLimit))
+            {
+                fileSizeLimit = int.MaxValue;
+            }
+
+            _fileSizeLimit = fileSizeLimit;
+
+            _fileSystemConfigurator = fileSystemConfigurator;
+
+            _allowedExtensions = configuration.GetSection("file:allowedextensions").Get<IEnumerable<string>>()?? new List<string>();
+        }
+        public async Task<FileMetadata> GetFileAsync(Guid fileId, int companyId)
+        {
+            var fileInfo = await _fileSystemConfigurator.GetConfigurationAsync(companyId, fileId);
+
+            if (fileInfo != null)
+            {
+                return fileInfo;
+            }
+            else
+            {
+                throw new RESTException($"File ID = {fileId} is not found for company = {companyId}");
+            }
+        }
+
+        public async IAsyncEnumerable<FileMetadata> GetAllFilesAsync(int companyId)
+        {
+            await foreach (var file in _fileSystemConfigurator.GetAllConfigurationsAsync(companyId))
+            {
+                yield return file;
+            }
+        }
+
+        public async Task<bool> RemoveFileAsync(Guid fileId, int companyId)
+        {
+            var fileInfo = await _fileSystemConfigurator.GetConfigurationAsync(companyId, fileId);
+
+            if(fileInfo != null)
+            {
+                if (File.Exists(fileInfo.FilePath))
+                {
+                    File.Delete(fileInfo.FilePath);
+
+                    await _fileSystemConfigurator.RemoveConfigurationAsync(companyId, fileId);
```

Comment:
first remove configuration and when you succeed then delete the file physically

Reply:
Probably some job in the future should be written to synchronize configuration file and file system content.